pandas/indexes/numeric.py

```diff
@@ -90,27 +93,44 @@ def _assert_safe_casting(cls, data, subarr):
         pass
 
 
-class Int64Index(NumericIndex):
-    """
+_num_index_shared_docs['class_descr'] = """
     Immutable ndarray implementing an ordered, sliceable set. The basic object
-    storing axis labels for all pandas objects. Int64Index is a special case
-    of `Index` with purely integer labels. This is the default index type used
-    by the DataFrame and Series ctors when no explicit index is provided by the
-    user.
+    storing axis labels for all pandas objects. %(klass)s is a special case
+    of `Index` with purely integer labels. %(extra)s
 
     Parameters
     ----------
     data : array-like (1-dimensional)
-    dtype : NumPy dtype (default: int64)
+    dtype : NumPy dtype (default: %(dtype)s)
     copy : bool
         Make a copy of input ndarray
     name : object
         Name to be stored in the index
 
     Notes
     -----
-    An Index instance can **only** contain hashable objects
-    """
+    An Index instance can **only** contain hashable objects.
+"""
+
+_num_index_shared_docs['_convert_scalar_indexer'] = """
+    Convert a scalar indexer.
```

Comment:
also pull out this doc-string to a separate PR. Idea is we want to do changes to make things generic in a PR before this one. Then this PR becomes much less complex.